packages/driver/cypress/integration/commands/sessions/manager_spec.ts

```diff
@@ -0,0 +1,355 @@
+      expect(sessionsSpy).to.be.calledOnce
+    })
+  })
+
+  // TODO
+  describe('._setStorageOnOrigins()', () => {})
+
+  it('.getAllHtmlOrigins()', async () => {
+    const storedOrigins = {
+      'https://example.com': {},
+      'https://foobar.com': {},
+    }
+
+    storedOrigins[`${baseUrl}`] = {}
+    const cypressSpy = cy.stub(CypressInstance, 'backend').callThrough().withArgs('get:rendered:html:origins').resolves(storedOrigins)
+    const sessionsManager = new SessionsManager(CypressInstance, cy)
+
+    const origins = await sessionsManager.getAllHtmlOrigins()
+
+    expect(cypressSpy).have.been.calledOnce
+    expect(origins).to.have.lengthOf(3)
+    expect(origins).to.deep.eq(['https://example.com', 'https://foobar.com', baseUrl])
+  })
+
+  describe('.sessions', () => {
+    it('sessions.defineSession()', () => {
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+      const sessionsSpy = cy.stub(sessionsManager, 'setActiveSession')
+      const setup = cy.stub()
+      const sess = sessionsManager.sessions.defineSession({ id: '1', setup })
+
+      expect(sess).to.deep.eq({
+        id: '1',
+        setup,
+        validate: undefined,
+        cookies: null,
+        localStorage: null,
+        hydrated: false,
+      })
+
+      expect(sessionsSpy).to.be.calledOnce
+      expect(sessionsSpy.getCall(0).args[0]).to.deep.eq({ 1: sess })
+    })
+
+    it('sessions.clearAllSavedSessions()', async () => {
+      const cypressSpy = cy.stub(CypressInstance, 'backend').withArgs('clear:session').resolves(null)
+
+      const sessionsManager = new SessionsManager(CypressInstance, () => {})
+      const sessionsSpy = cy.stub(sessionsManager, 'clearActiveSessions')
+
+      await sessionsManager.sessions.clearAllSavedSessions()
+
+      expect(sessionsSpy).to.be.calledOnce
+      expect(cypressSpy).to.be.calledOnceWith('clear:session', null)
+    })
+
+    it('.clearCurrentSessionData()', async () => {
+      // Unable to cleanly mock localStorage or sessionStorage on Firefox,
+      // so add dummy values and ensure they are cleared as expected.
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1141698
+      window.localStorage.foo = 'bar'
+      window.sessionStorage.jazzy = 'music'
+
+      expect(window.localStorage).of.have.lengthOf(1)
+      expect(window.sessionStorage).of.have.lengthOf(1)
+
+      const sessionsManager = new SessionsManager(CypressInstance, () => {})
+
+      const clearStorageSpy = cy.stub(sessionsManager.sessions, 'clearStorage')
+      const clearCookiesSpy = cy.stub(sessionsManager.sessions, 'clearCookies')
+
+      await sessionsManager.sessions.clearCurrentSessionData()
+
+      expect(clearStorageSpy).to.be.calledOnce
+      expect(clearCookiesSpy).to.be.calledOnce
+      expect(window.localStorage).of.have.lengthOf(0)
+      expect(window.sessionStorage).of.have.lengthOf(0)
+    })
+
+    // TODO
```

Comment:
I wonder if it is good to add the colon here to help some of the vscode plugins pick up the comment?
```suggestion
    // TODO:
```